File: BankApplication.java
/**
 * @author Rhiana Thelemaque
 */


/**
 * Import packages
 */
import java.util.Scanner;

public class BankApplication{
    public static void main(String[] args) {
        BankAccount obj = new BankAccount ("Rhiana", "A0108");
        obj1.showMenu();
    }
}

class BankAccount{
    int balance;
    int previousTransaction;
    String customerName;
    String customerId;

    // Bank account information
    BankAccount(String cname, String cid){
        customerName = cname;
        customerId = cid;
    }

    // Updating the account with the deposited amount.
    void deposit(int amount){
        // Only perform this if amount entered is not zero.
        if (amount != 0){
            balance += amount;
            previousTransaction = amount;
        }
    }

    // Updating the account with the withdrawn amount.
    void withdraw(int amount){
        // Only perform this if amount entered is not zero.
        if (amount != 0){
            balance -= amount;
            previousTransaction = -amount;
        }
    }

    // Shows the account holder the amount deposited or withdrawn in the most recent transaction.
    void getPreviousTransaction(){
        if (previousTransaction > 0){
            // If the account holder has deposited
            System.out.println("Amount deposited: " + previousTransaction);
        } else if (previousTransaction < 0){
            // If the account holder has withdrawn
            System.out.println("Amount withdrawn: " + Math.abs(previousTransaction));
        } else{
            // If there has been no transaction
            System.out.println("There was no transaction.");
        }        
    }

    // Displaying welcome message and menu
    void showMenu(){
        char option = '\0';
        Scanner scanner = new Scanner(System.in);

        System.out.println("Welcome " + customerName);
        System.out.println("Your ID is " + customerId); 
        System.out.println("\n");
        System.out.println("A. Check Balance.");
        System.out.println("B. Deposit.");
        System.out.println("C. Withdraw.");
        System.out.println("D. Previous Transaction.");
        System.out.println("E. Exit.");

        // Account holder selects option from the above 
        do{
            System.out.println("=============================================================");
            System.out.println("Enter an option: ");
            System.out.println("=============================================================");
            option = scanner.next().charAt(0);
            System.out.println("\n");

            switch(option){
                // Check balance
                case 'A':
                    System.out.println("-----------------------------------------------------");
                    System.out.println("Account balance: " + balance);
                    System.out.println("-----------------------------------------------------");
                    System.out.println("\n");
                    break;
                
                // Deposit
                case 'B':
                    System.out.println("-----------------------------------------------------");
                    System.out.println("Enter amount to deposit: ");
                    System.out.println("-----------------------------------------------------");
                    int amount = scanner.nextInt();
                    deposit(amount);
                    System.out.println("\n");
                    break;
                
                // Withdraw
                case 'C':
                    System.out.println("-----------------------------------------------------");
                    System.out.println("Enter amount to withdraw: ");
                    System.out.println("-----------------------------------------------------");
                    int amountTwo = scanner.nextInt();
                    deposit(amountTwo);
                    System.out.println("\n");
                    break;
                
                // Previous transaction
                case 'D':
                    System.out.println("-----------------------------------------------------");
                    getPreviousTransaction();
                    System.out.println("-----------------------------------------------------");
                    System.out.println("\n");
                    break;
                
                // If input is not in the given options
                default:
                    System.out.println("Invalid selection! Please enter another option.");
                    break;

            }

        } while (option != 'E');

        System.out.println("Thank you for using our services!");
    }
}

File: web app [in progress]/bankApplication.py




class BankAccount:
    def __init__(self) -> None:
        self.balance = 0
        self.previousTransaction = 0
        self.customerName = ''
        self.customerId = ''


    def account(self, cname, cid):
        self.customerName, self.customerId = cname, cid
    
    def deposit(self, amount):
        # if the amount is a positive integer
        if amount != 0:
            self.balance += 1
            self.previousTransaction = amount 
    
    def withdraw(self, amount):
        # if the amount is a positive integer
        if amount != 0:
            self.balance -= amount
            self.previousTransaction = -amount
    
    def getPreviousTransaction(self):
        if self.previousTransaction > 0:
            # if account holder deposited
            return 'Amount deposited: ' + self.previousTransaction
        elif self.previousTransaction < 0:
            # if account holder withdrew
            return 'Amount withdrawn: ' + abs(self.previousTransaction)
        else:
            # if there is no transaction
            return 'There was no transaction.'
    
    def showMenu(self):
        print('Welcome ' + self.customerName)
        print('Your ID is ' + self.customerId)
        print('\n')
        print('A. Check Balance.')
        print('B. Deposit.')
        print('C. Withdraw.')
        print('D. Previous Transaction.')
        print('E. Exit.')
        print('=============================================================')
        option = input('Enter an option: ').upper()
        print('=============================================================')
        print('\n')

        while option != 'E':
            # Check Balance
            if option == 'A':
                print('-----------------------------------------------------')
                print('Account balance: ' + self.balance)
                print('-----------------------------------------------------')


        return 'Thank you for using our services!'